docs/csharp/language-reference/operators/index.md

```diff
@@ -37,7 +37,8 @@ The following table lists the C# operators starting with the highest precedence
 | Operators | Category or name |
 | --------- | ---------------- |
 | [x.y](member-access-operators.md#member-access-operator-), [x?.y](member-access-operators.md#null-conditional-operators--and-), [x?[y]](member-access-operators.md#null-conditional-operators--and-), [f(x)](member-access-operators.md#invocation-operator-), [a&#91;i&#93;](member-access-operators.md#indexer-operator-), [x++](arithmetic-operators.md#increment-operator-), [x--](arithmetic-operators.md#decrement-operator---), [new](new-operator.md), [typeof](type-testing-and-cast.md#typeof-operator), [checked](../keywords/checked.md), [unchecked](../keywords/unchecked.md), [default](default.md), [nameof](nameof.md), [delegate](delegate-operator.md), [sizeof](sizeof.md), [stackalloc](stackalloc.md), [x->y](pointer-related-operators.md#pointer-member-access-operator--) | Primary |
-| [+x](arithmetic-operators.md#unary-plus-and-minus-operators), [-x](arithmetic-operators.md#unary-plus-and-minus-operators), [\!x](boolean-logical-operators.md#logical-negation-operator-), [~x](bitwise-and-shift-operators.md#bitwise-complement-operator-), [++x](arithmetic-operators.md#increment-operator-), [--x](arithmetic-operators.md#decrement-operator---), [(T)x](type-testing-and-cast.md#cast-operator-), [await](await.md), [&x](pointer-related-operators.md#address-of-operator-), [*x](pointer-related-operators.md#pointer-indirection-operator-), [true and false](true-false-operators.md) | Unary |
+| [+x](arithmetic-operators.md#unary-plus-and-minus-operators), [-x](arithmetic-operators.md#unary-plus-and-minus-operators), [\!x](boolean-logical-operators.md#logical-negation-operator-), [~x](bitwise-and-shift-operators.md#bitwise-complement-operator-), [++x](arithmetic-operators.md#increment-operator-), [--x](arithmetic-operators.md#decrement-operator---), [^x](member-access-operators.md#index-operator-), [(T)x](type-testing-and-cast.md#cast-operator-), [await](await.md), [&x](pointer-related-operators.md#address-of-operator-), [*x](pointer-related-operators.md#pointer-indirection-operator-), [true and false](true-false-operators.md) | Unary |
+| [x..y](member-access-operators.md#range-operator-) | Range |
```

Comment:
Given that `1..2*2` doesn't compile (cannot multiply range and `int`) and one needs parentheses `1..(2*2)`, I conclude that the `..` operator has a higher precedence than multiplication. Thus, the new separate row in the table.

Reply:
That is correct.

Reply:
@BillWagner with regards to the precedence level of the `..` operator, should the feature proposal note be updated?

Currently, [it says](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/ranges.md#systemrange):

> We replace the C# grammar rules for `shift_expression` with the following (in order to introduce a new precedence level):
>
> ```antlr
> shift_expression
>    : range_expression
>    | shift_expression '<<' range_expression
>    | shift_expression right_shift range_expression
>    ;
>
> range_expression
>    : additive_expression
>    | range_expression? '..' additive_expression?
>    ;
>```

Does that definition imply that addition has a higher precedence than the `..` operator? Or my understanding is not correct?

With the final implementation, shouldn't updated grammar rules look like:

```antlr
range_expression
  : unary_expression
  | range_expression? `..` unary_expression?   // OR: unary_expression? `..` unary_expression?

multiplicative_expression
    : range_expression
    | multiplicative_expression '*' range_expression
    | multiplicative_expression '/' range_expression
    | multiplicative_expression '%' range_expression
    ;
```

By the way, the [`OperatorPrecedence` enum](https://github.com/dotnet/roslyn/blob/master/src/Workspaces/CSharp/Portable/Extensions/OperatorPrecedence.cs) doesn't include any new precedence level for the `..` operator.

/cc @jcouv @agocke

Reply:
@pkulikov I had an offline discussion, and this part of the spec should be changing.  The updated precedence is:

> all forms of the .. operator have the same precedence, which is between unary and multiplicative, and that they are left-associative.

I'm working with the team to get the spec updated.